code/game/objects/items/devices/radio/encryptionkey.dm

```diff
@@ -22,6 +22,11 @@
 	channels = list("Syndicate" = 1)
 	origin_tech = "syndicate=1;engineering=3;bluespace=2"
 	syndie = 1 //Signifies that it de-crypts Syndicate transmissions
+	var/fake_name = "Agent ALERT_A_CODER"
+
+/obj/item/encryptionkey/syndicate/New()
+	..()
+	fake_name = "Agent " + pick(GLOB.html_colors)
```

Comment:
I have considered that, however since the list is global it is possible it might get used in other places. 

I can't just start deleting color names from there.

Reply:
could just have it copy the list, and the take from that list... I do agree with Bird I think

Reply:
Sure, I'll see how I can implement it.